### PR TITLE
feat: [barman] add init containers

### PR DESCRIPTION
**Pull request by kfirfer**
Added initContainers capability
For example waiting for postgresql instance(s)

### CODE REVIEW

File: charts/barman/Chart.yaml [hunk 1]
```diff
@@ -2,7 +2,7 @@ apiVersion: v2
 name: barman
 type: application
 description: Chart for Barman PostgreSQL Backup and Recovery Manager
-version: 0.9.1
```
Comment (eyenx) [inline charts/barman/Chart.yaml:5]: Thanks for your contribution!

As this is an additional feature, please bump to 0.10.0

**Review by eyenx — CHANGES_REQUESTED** [review]
Thanks for the contribution!

Please do a higher Chart version bump and please run pre-commit for generating the docs.

**Comment by eyenx** [comment]
For the pre-commit hook pls see https://github.com/adfinis/helm-charts/blob/main/docs/development.md

File: charts/barman/Chart.yaml (hunk 1 above)
Comment (eyenx) [reply charts/barman/Chart.yaml:5]: We are also missing the `artifacthub.io/changes` annotations in the Chart.yaml

See https://github.com/adfinis/helm-charts/pull/901 for a good example.

Thank you!

File: charts/barman/values.yaml [hunk 2]
```diff
@@ -32,6 +32,20 @@ deployment:
   imagePullSecrets: []
   # -- Specify additional ENVs for the Deployment
   additionalENVs: {}
+  # Init containers to add to deployment's pod spec. At least one plugin provider image is required.
+  # If the value is a string then it is evaluated as a template.
```
Comment (eyenx) [inline charts/barman/values.yaml:36]:
```suggestion
  # If the value is a string then it is evaluated as a template.
  # -- Specify initContainers for the deployment
```

Comment (eyenx) [reply charts/barman/values.yaml:36]: We need this to generate the README.md you modified manually.

Comment (kfirfer) [reply charts/barman/values.yaml]: New to helm docs
done (I guess)

File: charts/barman/templates/deployment.yaml [hunk 3]
```diff
@@ -81,6 +81,14 @@ spec:
           {{- end  }}
           resources:
             {{- toYaml .Values.resources | nindent 12 }}
+{{- if .Values.deployment.initContainers }}
+      initContainers:
+        {{- if eq (typeOf .Values.deployment.initContainers) "string" }}
```
Comment (eyenx) [inline charts/barman/templates/deployment.yaml:86]: Can you quickly explain why we would be needing using a template for setting the initContainers? Is it not enough just rendering the array toYaml?

I see this construct also in the official velero Chart.

Comment (kfirfer) [reply charts/barman/templates/deployment.yaml:86]: I just copied pasted from Velero chart
Don't know the reason for that
Should I use only array ?

Comment (kfirfer) [reply charts/barman/templates/deployment.yaml:86]: Hmm I guess I know the reason
if you're templating over templating(e.g. helmfile gotmpl files), and you already has a string based value in other charts

Comment (eyenx) [reply charts/barman/templates/deployment.yaml:86]: Makes sense

**Review by eyenx — APPROVED** [review]
LGTM thank you for your contribution!

**Comment by eyenx** [comment]
Release is here

https://github.com/adfinis/helm-charts/releases/tag/barman-0.10.0

**Pull request merged**